app/assets/javascripts/govuk_publishing_components/components/single-page-notification-button.js

```diff
@@ -0,0 +1,43 @@
+/* global XMLHttpRequest */
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function SinglePageNotificationButton ($module) {
+    this.$module = $module
+    this.personalisationEndpoint = this.$module.getAttribute('data-personalisation-endpoint')
```

Comment:
Why is it necessary to fetch the endpoint from a data attribute?   I see in the erb file you're generating this with the website root present, but doesn't it work when you use only the absolute path?